pyspark_ai/tool.py

```diff
@@ -101,12 +103,49 @@ async def _arun(self, *args: Any, **kwargs: Any) -> Any:
         raise NotImplementedError("ListTablesSqlDbTool does not support async")
 
 
+class LRUVectorStore:
+    """Implements an LRU policy to enforce a max storage space for vector file storage."""
+
+    def __init__(self, vector_file_dir: str, max_size: float = 1e6):
+        # by default, max_size = 1e6 bytes
+        self.cache = OrderedDict()
```

Comment:
We need to load the existing index files under the file path into the OrderedDict. The files under vector_file_dir can be reused after restarting PySparkAI.
(You can also refer some of the code from https://chat.openai.com/share/2bd66613-5dc1-427e-9277-fb45b317438a)

Reply:
Note: the implementation works best with a single thread. It's hard to support such a system with multiple threads anyway.